web/api/backend.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	jwtmiddleware "github.com/auth0/go-jwt-middleware"
+	"github.com/dgrijalva/jwt-go"
+
+	"github.com/gorilla/mux"
+)
+
+// Claims object that will be encoded to a JWT.
+// We add jwt.StandardClaims as an embedded type, to provide fields like expiry time
+type Claims struct {
+	UploadCredentials map[string]interface{}
+	jwt.StandardClaims
+}
+
+var jwtKey = []byte(os.Getenv("SECRET_KEY"))
+var creds WrapperCommand
+
+// TokenHandler to create the Token
+func TokenHandler(w http.ResponseWriter, r *http.Request) {
+	setupResponse(&w, r)
+	if (*&r).Method == "OPTIONS" {
+		return
+	}
+
+	// Get the JSON body and decode into credentials
+	err := json.NewDecoder(r.Body).Decode(&creds)
+	if err != nil {
+		// If the structure of the body is wrong, return an HTTP error
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if len(creds.Credential) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		io.WriteString(w, `{"error":"cannot make token without credentials"}`)
+		return
+	}
+
+	// Declare the expiration time of the token
+	// here, we have kept it as 5 minutes
+	expirationTime := time.Now().Add(1440 * time.Minute)
+	// Create the JWT claims, which includes the username and expiry time
+	claims := &Claims{
+		UploadCredentials: creds.Credential,
+
+		StandardClaims: jwt.StandardClaims{
+			// In JWT, the expiry time is expressed as unix milliseconds
+			ExpiresAt: expirationTime.Unix(),
+		},
+	}
+
+	// Declare the token with the algorithm used for signing, and the claims
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
```

Comment:
Again, I know you probably used the name token here because JWT is a token, but think about what this variable should be called in the context of your playground webapp so people don't confuse it with access tokens. For example maybe use "secureCredentials" or "signedCredentials" in your JSON output field instead of "token".

Reply:
Fixed!